routes/data.js

```diff
@@ -1,52 +1,127 @@
-const {Firestore} = require('@google-cloud/firestore');
+const { Firestore } = require('@google-cloud/firestore');
 
 // Database Initialization
-const firestore = new Firestore();
-const DB_COLLECTION_NAME = process.env.NODE_ENV == 'production' ? 'movies' : 'dev-movies';
-
-exports.view = function (req, res) {
-  createMovies();
-  const movies = readMovies();
-  movies.then((data) => {
-    res.send(data);
-  });
-};
+let firestore = new Firestore();
 
 /**
- * Inserts 3 new documents into the 'movies' collection with hard-coded data.
+ * Utility function that returns the correct collection name depending on whther the environment is PRODUCTION or DEVELOPMENT.
+ * @param {String} collectionName
  */
-async function createMovies() {
-  // Dummy data to insert into the database
-  const movieData = [
-    {
-      title: 'Black Panther',
-      year: 2018,
-    },
-    {
-      title: 'Avengers: Endgame',
-      year: 2019,
-    },
-    {
-      title: 'Iron Man',
-      year: 2008,
-    },
-  ];
-
-  movieData.forEach((movie) => {
-    const document = firestore.collection(DB_COLLECTION_NAME).doc();
-    document.set(movie);
-  });
+function resolveCollectionName(collectionName) {
+  return process.env.NODE_ENV == 'production'
+    ? collectionName
+    : 'dev-' + collectionName;
 }
 
+exports.setDatabase = function (firestoreInstance) {
+  firestore = firestoreInstance;
+};
+
+exports.getDatabase = function (firestoreInstance) {
+  return firestore;
+};
+
 /**
- * Read all documents in the 'movies' collection
+ * Queries for all AcceptedCategories by either organization or category depending on the  document reference and fieldName
```

Comment:
nit: remove extra space
```suggestion
 * Queries for all AcceptedCategories by either organization or category depending on the document reference and fieldName
```